src/foam/dao/EasyDAO.js

```diff
@@ -356,6 +364,15 @@ foam.CLASS({
         if ( getPm() )
           delegate = new foam.dao.PMDAO.Builder(getX()).setNSpec(getNSpec()).setDelegate(delegate).build();
 
+        if ( indexes.size() > 0 ) {
+          Object result = delegate.cmd_(getX(), new AddIndexCommand(false, indexes.toArray(new PropertyInfo[indexes.size()])));
```

Comment:
I think this will create one index with each component being what was intended to be a different index. One line 1131 below the PropertyInfo... is an array of parts of an index, but it looks like here each element is meant to be a different index.

Reply:
This retains existing behaviour and 'protected' and used to capture indexes from Aware interfaces encountered during delegate creation, and adding them at then end of delegate creation.